### ISSUE TITLE
Improvement for PostgreSQL connection options

### ISSUE THREAD

**Issue by schmandr** [opengisch/solocator#31]
I actually think things mentioned in #30 could be simplified if we solved it like in the "Create a new PostGIS connection" dialog in QGIS: ![postgis_connection](https://user-images.githubusercontent.com/3601534/67018778-fdfa2f00-f0fb-11e9-924c-ae8dfd23ef5d.png)
It contains input fields for _Service_, _Host_, _Port_, and the _Database_. So instead of forcing the user to choose between connecting using a service name or using host/port/database, we could leave it up to the user which fields to provide.

Just providing the service name and leaving the other fields emtpy (this must be the default) is perfectly fine. Then, an advanced user could add any of _Host_, _Port_, or _Database_ which would override the values stored in the PostgreSQL service file (which is the QGIS default behaviour). Or he could leave the Service name empty and just specify _Host_, _Port_, and _Database_ for defining a completely independent DB connection.

(Opposed to the standard QGIS dialog, for SoLocator we would omit the "Name" and "SSL mode" fields, and the "Port" field should be empty by default.)

What do you think of it?

**Comment by 3nids** [comment]
I think the plugin benefits from a simple default configuration which is possible because end users work in a predefined environment. 

At the origin there was only the auth config. PG service were added to give more freedom. 
Before making the configuration too complete, it would be good to define use cases to identify the different possible connection scenarios.

**Comment by andreasneumann** [comment]
Use case:
- be able to load tables from different PG servers (integration and test) where all of them have the same database setups
- the normal user doesn't have to use this. It is more for the employees of the SoGIS team who should be able to also use the integration and test environments

**Comment by 3nids** [comment]
therefore, a hidden setting would be better appropriate no?
we have this already, see https://github.com/opengisch/solocator/blob/master/README.md

**Comment by 3nids** [comment]
> We need to be able to have combinations of servicename (e.g. pub|edit|test) and hostnames (e.g. geodb|geodb-t|geodb-i)

How? A service usually contains the host, no? Or by servicename, you meant DB name?

> So I guess we would need an additional hidden "pg_host" setting, where the default is set to geodb.

At the moment, we have
* service name configurable in dialog
* hostname configurable by hidden setting
* DB name not configurable

The credentials can be given via configurable authcfg or through the service.

So, questions are:
1) What should be configurable?
2) Where?
  a) hidden setting 
  b) advanced config in the options
  c) layer loading options (CTRL+click)


It is there already.

**Comment by andreasneumann** [comment]
> How? A service usually contains the host, no? Or by servicename, you meant DB name?

A service configuration can contain any part or combination of a connection string, incl. authentication information.

If a servicename is provided and in addition, a definition of a connection string, that latter one can overwrite the definition in the service definition.

I did not know about this either before I started in Solothurn, but it is really totally flexible, but also makes it a bit difficult UI wise.

I'll talk again to @schmandr - but I think part of the configuration that only concerns admin or test users can go into the hidden settings.
